cli/tsc.rs

```diff
@@ -726,33 +731,62 @@ impl TsCompiler {
     let cwd = std::env::current_dir().unwrap();
     let performance =
       matches!(global_state.flags.log_level, Some(Level::Debug));
+    let unstable = self.flags.unstable;
+
+    let mut lib = if target == "main" {
+      vec!["deno.window"]
+    } else {
+      vec!["deno.worker"]
+    };
+
+    if unstable {
+      lib.push("deno.unstable");
+    }
+
+    let mut compiler_options = json!({
+      "allowJs": true,
+      "allowNonTsExtensions": true,
+      "checkJs": false,
+      "esModuleInterop": true,
+      "inlineSourceMap": false,
+      "jsx": "react",
+      "lib": lib,
+      "module": "system",
+      "outFile": "deno:///bundle.js",
+      // disabled until we have effective way to modify source maps
+      "sourceMap": false,
+      "strict": true,
+      "removeComments": true,
+      "target": "esnext",
+    });
 
     let compiler_config = self.config.clone();
 
-    // TODO(bartlomieju): this is non-sense; CompilerConfig's `path` and `content` should
-    // be optional
-    let j = match (compiler_config.path, compiler_config.content.as_ref()) {
-      (Some(config_path), Some(config_data)) => json!({
-        "type": msg::CompilerRequestType::Bundle,
-        "target": target,
-        "rootNames": root_names,
-        "unstable": self.flags.unstable,
-        "performance": performance,
-        "configPath": config_path,
-        "config": config_data.to_string(),
-        "cwd": cwd,
-        "sourceFileMap": module_graph_json,
-      }),
-      _ => json!({
-        "type": msg::CompilerRequestType::Bundle,
-        "target": target,
-        "rootNames": root_names,
-        "unstable": self.flags.unstable,
-        "performance": performance,
-        "cwd": cwd,
-        "sourceFileMap": module_graph_json,
-      }),
-    };
+    let maybe_ignored_options =
+      if let Some(config_text) = compiler_config.content.as_ref() {
+        let (user_config, ignored_options) =
+          tsc_config::parse_config(config_text)?;
+        tsc_config::json_merge(&mut compiler_options, &user_config);
+        ignored_options
+      } else {
+        None
+      };
+
+    warn_ignored_options(
+      maybe_ignored_options,
+      compiler_config.path.as_ref().unwrap(),
+    );
+
+    let j = json!({
+      "type": msg::CompilerRequestType::Bundle,
+      "target": target,
+      "rootNames": root_names,
+      "performance": performance,
+      "configPath": compiler_config.path.unwrap_or_else(|| PathBuf::from("tsconfig.json")),
```

Comment:
We can in theory just drop this and fake it always... it simply means that we don't support project references or extends, which we never supported anyways, and if we do, we would do it in Rust anyways and provide a single set of options anyways.

This is what I did in the rewrite branch: https://github.com/kitsonk/deno/blob/deno_compiler/compiler/compiler.js#L187-L194

Reply:
I tried, see comment in `parseCompilerOptions` method. I agree we should remove it completely.

Reply:
Actually already did